vcpkg.cmake

```diff
@@ -0,0 +1,21 @@
+###############################################################################
+## Copyright (C) 2023 Julian Xhokaxhiu <https://julianxhokaxhiu.com>
+##
+## This program is free software: you can redistribute it and/or modify
+## it under the terms of the GNU General Public License as published by
+## the Free Software Foundation, either version 3 of the License, or
+## (at your option) any later version.
+##
+## This program is distributed in the hope that it will be useful,
+## but WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+## GNU General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with this program.  If not, see <http://www.gnu.org/licenses/>.
+###############################################################################
+
+# VCPKG Kickstart toolchain file
+if(WIN32)
+  include(C:/vcpkg/scripts/buildsystems/vcpkg.cmake)
```

Comment:
What I understand about vcpkg configuration in a CMake project is you need to find the toolchain file path by hand. This path depends on the user installation, so I discourage guessing the value here and let the user follow the README and set the CMAKE_TOOLCHAIN_FILE variable by himself.

https://learn.microsoft.com/fr-fr/vcpkg/examples/installing-and-using-packages#cmake

Reply:
For example on my machine my path is E:/Programs/vcpkg/scripts/buildsystems/vcpkg.cmake

Reply:
I was fought around this as well, but as you noticed I left a trail on the README to clone `vcpkg` on that path so it should match the project. Ideally though we could script something in here that could guess where vcpkg is based on the hints it leaves when you run `vcpkg integrate install`. I didn't want to work on it right now, but could be a good task for anyone who jumps here and wants to contribute I think.

How I left it for now is kind of auto-wiring itself if you follow the README steps.

Reply:
Perhaps somethign like 
```
if(NOT VCPKG_ROOT)
     set (VCPKG_ROOT .. ) 
endif()
include (VCPKG_ROOT) 
```
this way you can change it at config time while providing a default

Reply:
Yep, good hint. Let me push that change.

Reply:
I did better, I added vcpkg as a submodule and make it relative to this project and this will work on Linux, macOS and Windows just fine in the future. Atm the toolchain var is defined only in the vscode settings.json file, but you could make use of it on the CI too by simply doing `-DCMAKE_TOOLCHAIN_FILE=vcpkg/scripts/buildsystems/vcpkg.cmake`

Reply:
I am not sure I fully agree with this approach, I will read more documentation about vcpkg and good practices and I'm coming back to you with an answer

Reply:
After digging onto the documentation of vcpkg, I agree that a submodule provide the best approach here.